<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT
</h1>

<hr>

<h2>Analyst: Alanis Danica Pura Concha</h2>

In [1]:
import numpy as np
import pandas as pd

In [2]:
penguin = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguin

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytic</h1>

In [4]:
penguin.describe()


bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h2>Filtering a column</h2>

In [5]:
penguin[penguin['species'] == "Adelie"]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [6]:
penguin['sex']

0        male
1      female
2      female
3         NaN
4      female
        ...  
339      male
340    female
341      male
342      male
343    female
Name: sex, Length: 344, dtype: object

In [7]:
adelie_df = penguin[penguin['species'] == "adelie"]

In [8]:
penguin['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [9]:
adelie_df['body_mass_g'].mean

<bound method Series.mean of Series([], Name: body_mass_g, dtype: float64)>

In [10]:
penguin['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [11]:
penguin[penguin['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

<h2>Cleaning the Data Frame</h2>

In [12]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h4>What to do with all the NULL VALUES</h4>

<h2>What to do with all the NULL VALUES</h2>

In [13]:
penguin['bill_length_mm'].mean()

43.9219298245614

In [14]:
penguin['sex'].mode()

0    male
Name: sex, dtype: object

<h2>HOW TO FILLUP NULL VALUES</h2>

In [15]:
penguin['bill_length_mm'] = penguin['bill_length_mm'].fillna(penguin['bill_length_mm'].mean())

In [16]:
penguin.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Changing Data Types</h2>

In [17]:
penguin['species'] = penguin['species'].astype('category')
penguin['species'] = penguin['species'].astype('category')

In [18]:
penguin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                333 non-null    object  
 7   year               344 non-null    int64   
dtypes: category(1), float64(4), int64(1), object(2)
memory usage: 19.4+ KB


<h1>ACTIVITY 1</h1>
<hr>

<h1>HOW MANY FEMALE PENGUINS IS THERE IN THE ISLAND OF DREAM?</h1>

In [19]:
female = penguin.groupby(['species','island', 'sex'], observed=True).size()

In [20]:
female

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      28
           Torgersen  female    24
                      male      23
Chinstrap  Dream      female    34
                      male      34
Gentoo     Biscoe     female    58
                      male      61
dtype: int64

<h2>"Insight 1: The female penguin in the island of dream is 27"</h2>

<h1>IN WHAT YEAR HAS THE MOST NUMBER OF PENGUINS?</h1>

In [21]:
penguin['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h2>"Insight 2: THE MOST NUMBER OF PENGUINS IS IN THE YEAR 2009"</h2>

<h1>WHAT YEAR HAS THE MOST AND LEAST PENGUINS?</h1>

In [22]:
penguin_per_year = penguin.groupby("year").size()

In [23]:
penguin_per_year

year
2007    110
2008    114
2009    120
dtype: int64

<h2>"Insight 3: 2009 has the most number of penguins, And the least number of penguins is on the year 2007"</h2>

<h1>WHICH SEX HAS THE HIGHER NUMBER OF PENGUINS</h1>

In [24]:
num = penguin.groupby(["sex"]).size()

In [25]:
num

sex
female    165
male      168
dtype: int64

<h2>"Insight 4: Male has the highest number of penguins"</h2>

<h1>what is the overall total number of penguins in all islands?</h1>

In [26]:
adelie_df = penguin[penguin['species'] == "adelie"]

In [27]:
penguin['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h2>"insight 5: The total number of penguins in biscoe is 168, in Dream has 124, while in Torgersen is 52"</h2>

<h1>WHAT ISLAND HAS THE MOST PENGUINS IN YEAR 2007?</h1>

In [28]:
penguin_2007 = penguin[penguin["year"] == 2007]

In [29]:
penguin_2007

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7              181.0   
1       Adelie  Torgersen        39.50000           17.4              186.0   
2       Adelie  Torgersen        40.30000           18.0              195.0   
3       Adelie  Torgersen        43.92193            NaN                NaN   
4       Adelie  Torgersen        36.70000           19.3              193.0   
..         ...        ...             ...            ...                ...   
297  Chinstrap      Dream        48.50000           17.5              191.0   
298  Chinstrap      Dream        43.20000           16.6              187.0   
299  Chinstrap      Dream        50.60000           19.4              193.0   
300  Chinstrap      Dream        46.70000           17.9              195.0   
301  Chinstrap      Dream        52.00000           19.0              197.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
297       3400.0    male  2007  
298       2900.0  female  2007  
299       3800.0    male  2007  
300       3300.0  female  2007  
301       4150.0    male  2007  

[110 rows x 8 columns]

In [30]:
island_counts = penguin_2007.groupby("island").size()

In [31]:
island_counts 

island
Biscoe       44
Dream        46
Torgersen    20
dtype: int64

<h2>"Insight 6: The island that has the most penguins in 2007 is Dream "</h2>

<h1>What is the overall body mass of penguins in 2009?</h1>

In [32]:
penguins_2009 = penguin[penguin["year"] == 2009] 

In [33]:
total_mass = penguins_2009["body_mass_g"].sum()

In [34]:
total_mass

501025.0

<h2>"Insight 7: The overall body mass of penguins in 2009 is 501025.0"</h2>

<h1>WHAT SPECIE HAS THE LARGER NUMBER IN 2007? </h1>

In [35]:
penguins_2007 = penguin[penguin["year"] == 2007] 

In [36]:
penguin_2007

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000           18.7              181.0   
1       Adelie  Torgersen        39.50000           17.4              186.0   
2       Adelie  Torgersen        40.30000           18.0              195.0   
3       Adelie  Torgersen        43.92193            NaN                NaN   
4       Adelie  Torgersen        36.70000           19.3              193.0   
..         ...        ...             ...            ...                ...   
297  Chinstrap      Dream        48.50000           17.5              191.0   
298  Chinstrap      Dream        43.20000           16.6              187.0   
299  Chinstrap      Dream        50.60000           19.4              193.0   
300  Chinstrap      Dream        46.70000           17.9              195.0   
301  Chinstrap      Dream        52.00000           19.0              197.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
297       3400.0    male  2007  
298       2900.0  female  2007  
299       3800.0    male  2007  
300       3300.0  female  2007  
301       4150.0    male  2007  

[110 rows x 8 columns]

<h2>"Insight 8: The specie that has the larger number in 2007 is Chinstrap"</h2>

<h1>HOW MANY FEMALE PENGUINS IS THERE IN THE ISLAND OF BISCOE?</h1>

In [37]:
female = penguin.groupby(['species','island', 'sex'], observed=True).size()

In [38]:
female

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      28
           Torgersen  female    24
                      male      23
Chinstrap  Dream      female    34
                      male      34
Gentoo     Biscoe     female    58
                      male      61
dtype: int64

<h2>"Insight 9: The female penguins in Biscoe islands is 22"</h2>

<h1>What is the overall body mass of penguins in 2008?</h1>

In [39]:
penguins_2008 = penguin[penguin["year"] == 2008] 

In [40]:
ttl_mss = penguins_2008["body_mass_g"].sum()

In [41]:
ttl_mss

486400.0

<h2>Insight 10: The overall body mass of penguins in 2008 is 486400.0</h2>

<h1>What is the overall body mass of all specie?</h1>

In [50]:
all_bdymss = penguin['body_mass_g'].sum()

In [51]:
all_bdymss

1437000.0

<h2>Insight 11: The Total of Body mass of all penguins is 1437000.0</h2>

<h1>what is the overall total number of all specie?</h1>

In [ ]:
adelie_df = penguin[penguin['species'] == "adelie"]

In [61]:
penguin['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h2>Insight 12: According from the given data above, it shows that the total number of species from various island starts from Biscoe is 168, tyhen 124 from Dream, while Torgesen has a least number which is 52</h2>